src/databricks/labs/ucx/hive_metastore/grants.py

```diff
@@ -219,20 +222,27 @@ def _grants(
             any_file=any_file,
             anonymous_function=anonymous_function,
         )
-        object_type_normalization = {"SCHEMA": "DATABASE", "CATALOG$": "CATALOG"}
-        for row in self._fetch(f"SHOW GRANTS ON {on_type} {key}"):
-            (principal, action_type, object_type, _) = row
-            if object_type in object_type_normalization:
-                object_type = object_type_normalization[object_type]
-            if on_type != object_type:
-                continue
-            yield Grant(
-                principal=principal,
-                action_type=action_type,
-                table=table,
-                view=view,
-                database=database,
-                catalog=catalog,
-                any_file=any_file,
-                anonymous_function=anonymous_function,
-            )
+        try:
+            object_type_normalization = {"SCHEMA": "DATABASE", "CATALOG$": "CATALOG"}
+            for row in self._fetch(f"SHOW GRANTS ON {on_type} {key}"):
+                (principal, action_type, object_type, _) = row
+                if object_type in object_type_normalization:
+                    object_type = object_type_normalization[object_type]
+                if on_type != object_type:
+                    logger.warning(
+                        f"Types for object {key} doesn't match after listing grants {object_type} != {on_type}"
+                    )
```

Comment:
```suggestion
                    logger.warning(
                        f"Types for object {key} doesn't match after listing grants {object_type} != {on_type}"
                    )
```
this will make a lot of noise if it's warning - there. either remove it or make it debug

Reply:
It's a corner case where the object type isn't the same as the one requested. For example you request the grant for a view and you get the grants for a table, it's not a normal behaviour and should be highlighted